ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitterSpec.scala

```diff
@@ -289,6 +293,76 @@ class TransactionCommitterSpec
     }
   }
 
+  "setting out of time bounds entry" should {
+
+    "be set" when {
+
+      "a submitting party is not known" in {
+        val context = createCommitContext(
+          recordTime = None,
+          inputs = createInputs(
+            Alice -> Some(hostedParty(Alice)),
+            Bob -> None,
+          ) + (Conversions.configurationStateKey -> None),
+          participantId = ParticipantId,
+        )
+        val transactionEntry = createEmptyTransactionEntry(List(Alice, Bob))
+        val result = transactionCommitter.preExecute(
+          DamlSubmission.newBuilder().setTransactionEntry(transactionEntry).build(),
+          context,
+        )
+        resultIsRejectedWithPayload(
+          result,
+          DamlTransactionRejectionEntry.ReasonCase.SUBMITTING_PARTY_NOT_KNOWN_ON_LEDGER,
+        )
+      }
+
+      "the command is a duplicate" in {
+        val transactionEntry = createEmptyTransactionEntry(List(Alice))
+        val context = createCommitContext(
+          recordTime = None,
+          inputs = createInputs(
+            Alice -> Some(hostedParty(Alice))
+          ) + (Conversions.configurationStateKey -> None) + (Conversions.commandDedupKey(
+            transactionEntry.getSubmitterInfo
+          ) -> Some(
```

Comment:
This makes the test case very hard to read. Can you refactor this so that the input is readable, i.e., preferably just a list of `key -> value` pairs, please?
E.g., moving the definition of the `commandDedupStateValue` out should improve it.